### ISSUE TITLE
if I want to use my image-text data to finetune the SigLIP2， where I can get the train code?

### ISSUE THREAD

**Issue by LeopardCatCat** [google-research/big_vision#172]

**Comment by mitscha** [comment]
You can fine-tune with the sigmoid loss/SigLIP training code using [this config](https://github.com/google-research/big_vision/blob/main/big_vision/configs/proj/image_text/siglip_lit_coco.py). You'll have to slightly adapt the model code (variant, pool_type, init with the pretrained weights), preprocessing function, schedule (the config freezes the image encoder), and your data set. You can look the details up in the [demo colab](https://colab.research.google.com/github/google-research/big_vision/blob/main/big_vision/configs/proj/image_text/SigLIP2_demo.ipynb).

Alternatively you can use the [HF Transformers models](https://huggingface.co/collections/google/siglip2-67b5dcef38c175486e240107) with your favorite fine-tuning codebase.

**Comment by z2587721154** [comment]
> You can fine-tune with the sigmoid loss/SigLIP training code using [this config](https://github.com/google-research/big_vision/blob/main/big_vision/configs/proj/image_text/siglip_lit_coco.py). You'll have to slightly adapt the model code (variant, pool_type, init with the pretrained weights), preprocessing function, schedule (the config freezes the image encoder), and your data set. You can look the details up in the [demo colab](https://colab.research.google.com/github/google-research/big_vision/blob/main/big_vision/configs/proj/image_text/SigLIP2_demo.ipynb).
> 
> Alternatively you can use the [HF Transformers models](https://huggingface.co/collections/google/siglip2-67b5dcef38c175486e240107) with your favorite fine-tuning codebase.

Thank you very much for your contribution to the open source community, the code about siglip2 fine-tuned on cat-seg, can it be made public?
